core/modules/drr.cc

```diff
@@ -274,21 +303,21 @@ void DRR::AddNewFlow(bess::Packet* pkt, FlowId id, int* err) {
   Flow* f = new Flow(id);
 
   // TODO(joshua) do proper error checking
-  f->queue = AddQueue(static_cast<int>(kFlowQueueSize), err);
+  f->queue = generator_();
```

Comment:
Given that there are only two types of Queue, I am somewhat concerned that the generator/factory mechanism might be a little over-engineered. Can we just simply do something like this...?
```
if (queue type is codel) {
  f->queue = Codel(codel params);
} if (queue type is llring) {
  f->queue = LockLessQueue(llring params);
}
```

This is less fancy, but shorter and simpler.